whois-api/src/main/java/net/ripe/db/whois/api/rdap/RdapObjectMapper.java

```diff
@@ -133,13 +138,15 @@ public RdapObjectMapper(
             final ReservedResources reservedResources,
             final Ipv4Tree ipv4Tree,
             final Ipv6Tree ipv6Tree,
-            @Value("${rdap.port43:}") final String port43) {
+            @Value("${rdap.port43:}") final String port43,
+            @Value("${rdap.public.baseUrl:}") final String baseUrl) {
         this.noticeFactory = noticeFactory;
         this.rpslObjectDao = rpslObjectDao;
         this.ipv4Tree = ipv4Tree;
         this.ipv6Tree = ipv6Tree;
         this.port43 = port43;
         this.reservedResources = reservedResources;
+        this.rdapRirSearchSkeleton = baseUrl + "/%s/rirSearch1/%s/%s";
```

Comment:
Define this structure in a static method using UriBuilder

Reply:
baseUrl come from the properties, it is not static.
I cannot make an static method using baseUrl

Reply:
OK. Then use a private method (not static) to generate the URL using UriBuilder (i.e. don't use String concatenation).

Reply:
Agreed to use UriBuilder